src/AddRecipeForm.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react"
+import UseInputState from "./Hooks/useInputState"
+import Button from "./Style/Button"
+import Input from "./Style/Input"
+import Label from "./Style/Label"
+import { v4 as uuid } from 'uuid';
+
+function AddRecipeForm({ addRecipe, toggle }) {
+    const [nameVal, nameChange, nameReset] = UseInputState("")
+    const [descVal, descChange, descReset] = UseInputState("")
+    const [newIngredient, newIngredientChange, newIngredientReset] = UseInputState("")
+    const [ingredients, setIngredients] = useState([])
+
+    const handleRemove = (e) => {
+        const newIngredients = ingredients.filter(ing => ing.name !== e.target.name)
+        setIngredients(newIngredients)
+    }
+
+    const handleAdd = () => {
+        if (newIngredient) {
+            setIngredients([...ingredients, { "name": newIngredient }])
+            newIngredientReset()
+        }
+    }
+
+    return <form onSubmit={e => {
+        e.preventDefault()
+        if (nameVal.trim() && descVal.trim() && ingredients.length > 0) {
```

Comment:
This line is a bit hard to understand, I'd separate it into variables, create smaller functions or add a comment to make it easier to understand